### PR TITLE
Added GenWQE tests

### PR DESCRIPTION
**Pull request by narasimhan-v**
Added GenWQE sanity tests

Signed-off-by: Narasimhan V <sim@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by narasimhan-v** [comment]
```
# avocado run genqwetest.py -m genqwetest.py.data/genwqetest.yaml 
JOB ID     : 6f3d2b7d44cd1255f22f357daa9298da53377b72
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2017-06-09T01.33-6f3d2b7/job.log
 (1/9) genqwetest.py:GenWQETest.test_genwqe_echo;run-0fb5: PASS (2.46 s)
 (2/9) genqwetest.py:GenWQETest.test_genwqe_mt_perf;run-0fb5: PASS (178.60 s)
 (3/9) genqwetest.py:GenWQETest.test_genwqe_test_gz;run-0fb5: PASS (46.34 s)
 (4/9) genqwetest.py:GenWQETest.test_genwqe_memcopy;run-0fb5: FAIL (18.75 s)
 (5/9) genqwetest.py:GenWQETest.test_genwqe_gzip_gunzip;run-0fb5: PASS (88.06 s)
 (6/9) genqwetest.py:GenWQETest.test_genwqe_poke;run-0fb5: PASS (10.91 s)
 (7/9) genqwetest.py:GenWQETest.test_genwqe_peek;run-0fb5: PASS (0.88 s)
 (8/9) genqwetest.py:GenWQETest.test_genwqe_ffdc;run-0fb5: PASS (0.91 s)
 (9/9) genqwetest.py:GenWQETest.test_genwqe_cksum;run-0fb5: FAIL (11.93 s)
RESULTS    : PASS 7 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 361.02 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2017-06-09T01.33-6f3d2b7/html/results.html
```

File: io/genwqe/genwqetest.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        Install genwqe packages, and downloads test tarball.
+        """
+        self.card = self.params.get('device', default='0')
+        url = "http://corpus.canterbury.ac.nz/resources/cantrbry.tar.gz"
+        self.url = self.params.get('test_tar_url', default=url)
+        if not os.path.isdir("/sys/class/genwqe/genwqe%s_card/" % self.card):
+            self.skip("Device %s does not exist" % self.card)
+        smm = SoftwareManager()
+        pkgs = ['genwqe-tools', 'genwqe-zlib']
+        for pkg in pkgs:
```
Comment (PraveenPenguin) [inline io/genwqe/genwqetest.py]: can this be like 
`for pkg in ['genwqe-tools', 'genwqe-zlib']:`

File: io/genwqe/genwqetest.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+            if not smm.check_installed(pkg) and not smm.install(pkg):
+                self.skip('pkg is needed for the test to be run')
```
Comment (PraveenPenguin) [inline io/genwqe/genwqetest.py]: can you please display package name as well .it help user which package is required ?

File: io/genwqe/genwqetest.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        if failures:
+            self.fail("%s fails on file comparison" % ", ".join(failures))
+
+    def test_genwqe_gzip_gunzip(self):
+        """
+        Tests genwqe_gzip and genwqe_gunzip on the device.
+        """
+        self.files_used.extend(["file", "file.gz"])
+        cmd = "dd if=/dev/urandom bs=4096 count=100000 of=file"
+        process.system(cmd, shell=True)
```
Comment (PraveenPenguin) [inline io/genwqe/genwqetest.py]: please use `ignore_status=True` as a parm

File: io/genwqe/genwqetest.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        for opt in ['', '-1', '-9']:
+            orig_size = os.stat("file").st_size
+            cmd = "genwqe_gzip -A GENWQE -B %s file %s" % (self.card, opt)
+            if process.system(cmd, shell=True, ignore_status=True):
+                self.fail("genwqe_gzip %s fails" % opt)
+            cmd = "genwqe_gunzip -A GENWQE -B %s file.gz %s" % (self.card, opt)
+            if process.system(cmd, shell=True, ignore_status=True):
+                self.fail("genwqe_gunzip %s fails" % opt)
+            processed_size = os.stat("file").st_size
+            if orig_size != processed_size:
```
Comment (PraveenPenguin) [inline io/genwqe/genwqetest.py]: can this be like 
`
if orig_size != os.stat("file").st_size`

File: io/genwqe/genwqetest.py (hunk 1 above)
Comment (narasimhan-v) [inline io/genwqe/genwqetest.py]: Done

File: io/genwqe/genwqetest.py (hunk 2 above)
Comment (narasimhan-v) [inline io/genwqe/genwqetest.py]: Done

File: io/genwqe/genwqetest.py (hunk 3 above)
Comment (narasimhan-v) [inline io/genwqe/genwqetest.py]: Done

File: io/genwqe/genwqetest.py (hunk 4 above)
Comment (narasimhan-v) [inline io/genwqe/genwqetest.py]: Have used the variables in the next lines

Comment (PraveenPenguin) [inline io/genwqe/genwqetest.py]: yup i already saw that i think both place can be have with this change again it is just a suggestion and it's your call , please let me know what you think

Comment (narasimhan-v) [inline io/genwqe/genwqetest.py]: I prefer the existing way please ;)

Comment (PraveenPenguin) [inline io/genwqe/genwqetest.py]: sure

**Comment by PraveenPenguin** [comment]
@narasimhan-v thanks

**Pull request merged**